src/cartridge/mod.rs

```diff
@@ -112,6 +133,61 @@ impl Cartridge {
         } else {
             self.header.cart_mode = CartridgeMode::PgbMode;
         }
+
+       self.header.cart_type = match self.mem_read_byte(0x147) {
+            0x00 => CartridgeType::RomOnly,
+            0x01 => CartridgeType::Mbc1,
+            0x02 => CartridgeType::Mbc1Ram,
+            0x03 => CartridgeType::Mbc1RamBattery,
+            0x05 => CartridgeType::Mbc2,
+            0x06 => CartridgeType::Mbc2Battery,
+            0x08 => CartridgeType::RomRam,
+            0x09 => CartridgeType::RomRamBattery,
+            0x0b => CartridgeType::Mmm01,
+            0x0c => CartridgeType::Mmm01Ram,
+            0x0d => CartridgeType::Mmm01RamBattery,
+            0x0f => CartridgeType::Mbc3TimerBattery,
+            0x10 => CartridgeType::Mbc3TimerRamBattery,
+            0x11 => CartridgeType::Mbc3,
+            0x12 => CartridgeType::Mbc3Ram,
+            0x13 => CartridgeType::Mbc3RamBattery,
+            0x19 => CartridgeType::Mbc5,
+            0x1a => CartridgeType::Mbc5Ram,
+            0x1b => CartridgeType::Mbc5RamBattery,
+            0x1c => CartridgeType::Mbc5Rumble,
+            0x1d => CartridgeType::Mbc5RumbleRam,
+            0x1e => CartridgeType::Mbc5RumbleRamBattery,
+            0x20 => CartridgeType::Mbc6,
+            0x22 => CartridgeType::Mbc7SensorRumbleRamBattery,
+            0xfc => CartridgeType::PocketCamera,
+            0xfd => CartridgeType::BandaiTama5,
+            0xfe => CartridgeType::HuC3,
+            0xff => CartridgeType::HuC1RamBattery,
+            _ => panic!("invalid cartridge type"),
+        };
+
+        let rom_size_code = self.mem_read_byte(0x148);
+        if rom_size_code > 0x08 {
+            panic!("invalid ROM size");
```

Comment:
I explicitly left it unsupported because of this:
```
Only listed in unofficial docs. No cartridges or ROM files using these sizes are known. As the other ROM sizes are all powers of 2, these are likely inaccurate. The source for these values is unknown.
```

Reply:
Well... that is a good point. Okay, let's leave out the weird shit.

Reply:
Added a comment for this in `b7169b4`